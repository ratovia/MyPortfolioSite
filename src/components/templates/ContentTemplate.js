import React from "react"
import styled from "styled-components"
import { Layout } from "../layout"

const Nav = styled.div`
  width: 108px;
  height: 100vh;
  position: fixed;
  top: 0;
  left: 0;
  z-index: 10;
`

const HeroContent = styled.div`
  display: flex;
  margin-left: 108px;
  width: calc(100vw - 108px);
  min-height: 100px;
  background-color: #414141;
`
const MainContent = styled.div`
  display: flex;
  margin-left: 108px;
  width: calc(100vw - 108px);
  min-height: 100px;
  background-color: #414141;
`

const FooterContent = styled.div`
  display: flex;
  margin-left: 108px;
  width: calc(100vw - 108px);
  min-height: 200px;
  background-color: #414141;
`

export const ContentTemplate = ({
  navigation,
  heroContent,
  mainContent,
  footerContent,
}) => {
  return (
    <Layout>
      <Nav>{navigation}</Nav>
      <HeroContent>{heroContent}</HeroContent>
      <MainContent>{mainContent}</MainContent>
      <FooterContent>{footerContent}</FooterContent>
    </Layout>
  )
}
