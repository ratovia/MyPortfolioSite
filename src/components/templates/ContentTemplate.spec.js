import React from "react"
import renderer from "react-test-renderer"
import { ContentTemplate } from "./ContentTemplate"
describe("ContentTemplate", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <ContentTemplate
          headerText="Hello World"
          navigation={
            <div
              style={{
                width: "100%",
                height: "100%",
                backgroundColor: "red",
              }}
            ></div>
          }
          heroContent={
            <div
              style={{
                width: "100%",
                minHeight: "100px",
                backgroundColor: "blue",
              }}
            ></div>
          }
          mainContent={
            <div
              style={{
                width: "100%",
                minHeight: "100px",
                backgroundColor: "yellow",
              }}
            ></div>
          }
          footerContent={
            <div
              style={{
                width: "100%",
                minHeight: "200px",
                backgroundColor: "orange",
              }}
            ></div>
          }
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
