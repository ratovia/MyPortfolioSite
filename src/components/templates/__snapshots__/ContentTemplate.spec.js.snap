// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentTemplate renders correctly 1`] = `
<div>
  <div
    className="sc-bdvvaa jLICYy"
  >
    <div
      style={
        Object {
          "backgroundColor": "red",
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </div>
  <div
    className="sc-gsDJrp dQjCJY"
  >
    <div
      style={
        Object {
          "backgroundColor": "blue",
          "minHeight": "100px",
          "width": "100%",
        }
      }
    />
  </div>
  <div
    className="sc-dkPtyc hugYLJ"
  >
    <div
      style={
        Object {
          "backgroundColor": "yellow",
          "minHeight": "100px",
          "width": "100%",
        }
      }
    />
  </div>
  <div
    className="sc-hKwCoD boTHiz"
  >
    <div
      style={
        Object {
          "backgroundColor": "orange",
          "minHeight": "200px",
          "width": "100%",
        }
      }
    />
  </div>
</div>
`;
